src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.Analyzer.cs

```diff
@@ -129,6 +130,72 @@ protected override bool ReadOnlyFieldAllowed()
                 var scope = this.SelectionResult.GetContainingScopeOf<ConstructorDeclarationSyntax>();
                 return scope == null;
             }
+
+            protected override ITypeSymbol GetSymbolType(SemanticModel semanticModel, ISymbol symbol)
+            {
+                var selectionOperation = semanticModel.GetOperation(this.SelectionResult.GetContainingScope());
+
+                switch (symbol)
+                {
+                    case ILocalSymbol local:
+                        if (local.NullableAnnotation == NullableAnnotation.Annotated)
+                        {
+                            var references = selectionOperation.DescendantsAndSelf()
+                                .Where(o => o switch
+                                {
+                                    ILocalReferenceOperation localReference => localReference.Local == symbol,
+                                    IAssignmentOperation assignment => assignment.Target is ILocalReferenceOperation localReference && localReference.Local == symbol,
+                                    _ => false
+                                });
+
+                            var flowState = GetFlowStateFromLocations(references, local.NullableAnnotation);
+                            if (flowState != NullableFlowState.None)
```

Comment:
Why the != None check?

Reply:
Maybe a misinterpretation of how the flow state works, but if for some reason flow state can't determine nullability does it use `None` or does it default to `MaybeNull`? If it uses `None` I don't want that to override the nullability, I want to just take the type with nullability as is

Reply:
The compiler maps oblivious to _not null_ because that prevents downstream warnings about using something that might be potentially null.

Reply:
@333fred could clarify if there's a case where we would get a None. I think my impression was that was mostly if you have a case which doesn't have a concept of flow state. Although maybe there's some weirdness around things that reference a thing that has flow state but itself is flow-agnostic. Something liken nameof() where it's using a variable, but the variable's state of any kind is pointless?

Reply:
Yeah, `None` in flow state should only be for places where nullability has no meaning, such as on a statement.